tests/cli/test_status_diff.py

```diff
@@ -0,0 +1,1188 @@
+        state_diff = {'interfaces': {'enp5s0': {'index': 2, 'name': 'enp5s0', 'id': 'enp5s0', 'system_state': {}, 'netplan_state': {}}}, 'missing_interfaces_system': {}, 'missing_interfaces_netplan': {'lo': {'type': 'ethernet', 'index': 1}}}  # nopep8
+
+        expected = '''+ ●  1: lo ethernet UNKNOWN/UP (unmanaged)
+          MAC Address: 00:00:00:00:00:00
+            Addresses: 127.0.0.1/8
+                       ::1/128
+               Routes: ::1 metric 256
+
+  ●  2: enp5s0 ethernet UP (networkd: enp5s0)
+          MAC Address: 00:16:3e:71:d0:1f (Red Hat, Inc.)
+            Addresses: 10.86.126.148/24 (dhcp)
+        DNS Addresses: 10.86.126.1
+           DNS Search: lxd
+               Routes: default via 10.86.126.1 from 10.86.126.148 metric 100 (dhcp)
+                       10.86.126.0/24 from 10.86.126.148 metric 100 (link)
+                       10.86.126.1 from 10.86.126.148 metric 100 (dhcp, link)
+
+Use "--diff-only" to omit the information that is consistent between the system and Netplan.
+'''
+
+        f = io.StringIO()
+        with redirect_stdout(f):
+            status = NetplanStatus()
+            status.ifname = None
+            status.verbose = False
+            status.diff = True
+            status.diff_only = False
+            status.state_diff = state_diff
+            status.pretty_print(input_data, 2, _console_width=130)
+            out = f.getvalue()
+            self.assertEqual(out, expected)
+
+        # Same test for --diff-only
+
+        expected = '''+ ●  1: lo ethernet UNKNOWN/UP (unmanaged)
+
+'''
```

Comment:
question (non-blocking): Should we trim unused whitespace in the output?

Reply:
I suppose it would be nice. It will require checking if the interface we are displaying is or is not the last one. I'll do that